File: app/src/main/java/com/yeraygarcia/recipes/viewmodel/RecipeDetailViewModelFactory.java
package com.yeraygarcia.recipes.viewmodel;

import android.arch.lifecycle.ViewModel;
import android.arch.lifecycle.ViewModelProvider;
import android.support.annotation.NonNull;

import com.yeraygarcia.recipes.database.repository.RecipeDetailRepository;

import java.util.UUID;

public class RecipeDetailViewModelFactory extends ViewModelProvider.NewInstanceFactory {
    private final RecipeDetailRepository mRepository;
    private final UUID mTaskId;

    public RecipeDetailViewModelFactory(RecipeDetailRepository repository, UUID taskId) {
        mRepository = repository;
        mTaskId = taskId;
    }

    @NonNull
    @Override
    public <T extends ViewModel> T create(@NonNull Class<T> modelClass) {
        //noinspection unchecked
        return (T) new RecipeDetailViewModel(mRepository, mTaskId);
    }
}

File: app/src/main/java/com/yeraygarcia/recipes/viewmodel/RecipeDetailViewModelFactory.kt
package com.yeraygarcia.recipes.viewmodel

import android.arch.lifecycle.ViewModel
import android.arch.lifecycle.ViewModelProvider
import com.yeraygarcia.recipes.database.repository.RecipeDetailRepository
import java.util.*

class RecipeDetailViewModelFactory(
    private val mRepository: RecipeDetailRepository,
    private val mTaskId: UUID
) : ViewModelProvider.NewInstanceFactory() {

    override fun <T : ViewModel> create(modelClass: Class<T>): T {
        @Suppress("UNCHECKED_CAST")
        return RecipeDetailViewModel(mRepository, mTaskId) as T
    }
}
